### PR TITLE
Makefile for pipeline

### PR DESCRIPTION
**Pull request by Docproc**
A Makefile, with targets for each stage of the pipeline, as well as some composite targets and utilities.

The Makefile uses two scripts in `scripts/`, one for checking the existence of required indices, and one for downloading a specified percentage of a remote file.

A section describing the use of the Makefile has been added to README.md.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,234 @@
+# Makefile for Open Targets data pipeline
+# Based on steps in https://github.com/opentargets/data_release/wiki/OT011-Data-Processing-(Backend)
+
+# THhese variables can be overridden on the command-line
+ES_PREFIX = gp-makefile
+ELASTICSEARCH_NODES = http://localhost:9200
```
Comment (mkarmona) [inline Makefile]: does it @Docproc have to get the `es URL` from `db.ini` file better than having two different places to change it?

Comment (Docproc) [reply Makefile]: I thought of that, but I don't think it's possible to easily get the value from the `db.ini` file into the Makefile, at least not without introducing additional dependencies (i.e. some sort of INI file parser). Is there an easy way I'm missing?

File: scripts/check_index.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                if es.count(index).get("count") == 0:
+                    exit_code = 1
+                    if not args.silent:
+                        print "Required index %s exists but has zero entries" % index
+        else:
+            exit_code = 1
+            if not args.silent:
+                print "Required index %s does not exist" % index
+
+    sys.exit(exit_code)
```
Comment (afaulconbridge) [inline scripts/check_index.py]: Shouldn't this be `return exit_code` so that the caller can handle the actual exit?

File: Makefile (hunk 1 above)
Comment (afaulconbridge) [reply Makefile]: We can also allow preset environment variables to override the ones inside the Makefile with `?=` syntax instead fo `=`as described https://stackoverflow.com/a/40241012

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+PERCENTAGE_TO_KEEP = 2
+
+# Internal variables
+MRTARGET_CMD = python -m mrtarget.CommandLine --log-level=$(LOG_LEVEL) $(ES_PREFIX)
+INDEX_CMD = python scripts/check_index.py --elasticsearch $(ELASTICSEARCH_NODES) --zero-fail --index
+DOWNLOAD_CMD = ./scripts/download_and_trim.sh
+
+# Main command for invoking mrtarget; note that $STAGE needs to be defined for this work.
+# Also note that $@ expands to the name of the current Makefile target so saves a lot of boilerplate
+TEMPLATE_CMD = $(MRTARGET_CMD) --$(STAGE) 2>&1 | tee out.$(ES_PREFIX).$(STAGE).log
```
Comment (afaulconbridge) [inline Makefile]: Can we add an ability to specify additional arguments to pass to stages e.g. `$(MRTARGET_ARGS)` ? Would be useful for the QC steps I'm working on.

Comment (Docproc) [reply Makefile]: Good idea, commit 68f59a8 which I just pushed allows this.

File: scripts/check_index.py (hunk 2 above)
Comment (Docproc) [reply scripts/check_index.py]: It should, commit 909b413 fixes this

File: Makefile (hunk 1 above)
Comment (Docproc) [reply Makefile]: I didn't know you could do that! Change made, see 0833d3f

**Pull request merged**